### PR TITLE
Return detailed msg for notification errors

### LINKED ISSUE #5908: Minio fails to start due to a Postgres notification  without saying the detailed error
## Expected Behavior
In the config,json file I setup a notification to a Postgres system. I am sure that the Postgres instance is running and reachable from the machine where Minio is running , the table exists and the credentials are correct. However when I try to start the Minio process it fails to start with the error "Error: Unable to initialize notification system". I would expect somehow to have a more descriptive error when Minio fails to start to troubleshoot the problem

## Current Behavior
There is no way to get a more descriptive and detailed explanation of the error if Minio fails to start for any reason (Postgres notification in this specific case). So it is impossible to troubleshoot the problem

## Possible Solution
Add a more detailed log with error information even when Minio starts to fail

## Steps to Reproduce (for bugs)
1. setup a Postgres notification in the config.json file specifying  not existing host to provoke the error
2. launch minio and check that the only error information available is "Error: Unable to initialize notification system"

## Context

We want Minio to send notification to Postgres in order to process immediately data files submitted from external providers. Those file contain data that we want to store in our internal database.

## Your Environment

* Version used (`minio version`): 23
* Server type and version: Linux Redhat
* Operating System and version (`uname -a`): Linux 3.10.0-514.16.1.el7.x86_64 #1 SMP Fri Mar 10 13:12:32 EST 2017 x86_64 x86_64 x86_64 GNU/Linux
* Link to your project: http://europepmc.org

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
This PR will print detailed error messages for any notification error for now. In contrary to policy subsystem, notification errors can be legitimate due to users mis-configuration 

## Motivation and Context
Fixes #5908 

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/server-main.go [hunk 1]
```diff
@@ -246,13 +246,13 @@ func serverMain(ctx *cli.Context) {
 	var handler http.Handler
 	handler, err = configureServerHandler(globalEndpoints)
 	if err != nil {
-		logger.Fatal(uiErrUnexpectedError(err), "Unable to configure one of server's RPC services.")
+		logger.Fatal(uiErrUnexpectedError(err), "Unable to configure one of server's RPC services")
```
Comment (harshavardhana) [inline cmd/server-main.go:249]: why remove the `.` is it removed everywhere?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/server-main.go (hunk 1 above)
Comment (vadmeste) [reply cmd/server-main.go:249]: > why remove the . is it removed everywhere?

It is not consistent, I'll remove them in this PR.

Comment (vadmeste) [reply cmd/server-main.go:249]: > why remove the . is it removed everywhere?

Fixed

**Review by ebozduman — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#4dfcd67d-3ef1-443a-bb63-4be827a82166) |
<a name="4dfcd67d-3ef1-443a-bb63-4be827a82166"></a>
### 5913-81e37f7/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:32115
ACCESS_KEY:      minio
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp 936d8cb92d77:/mint/log /tmp/mint-logs'
(1/13) Running aws-sdk-go tests ... done in 4 seconds
(2/13) Running aws-sdk-java tests ... done in 23 seconds
(3/13) Running aws-sdk-php tests ... done in 57 seconds
(4/13) Running aws-sdk-ruby tests ... done in 53 seconds
(5/13) Running awscli tests ... done in 4 minutes and 22 seconds
(6/13) Running mc tests ... done in 3 minutes and 42 seconds
(7/13) Running minio-dotnet tests ... done in 1 minutes and 1 seconds
(8/13) Running minio-go tests ... done in 21 seconds
(9/13) Running minio-java tests ... done in 44 seconds
(10/13) Running minio-js tests ... FAILED in 45 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-1b8ca82a-f1ab-40ce-9466-5813e1a8da68, events: s3:ObjectCreated:*",
  "duration": 8808,
  "status": "FAIL",
  "error": "Error: notification not received at client.removeObject.err (test/functional-tests.js:1034:43) at ClientRequest.<anonymous> (node_modules/minio/dist/main/minio.js:420:9) at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:478:21) at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23) at Socket.socketOnData (_http_client.js:367:20) at readableAddChunk (_stream_readable.js:176:18) at Socket.Readable.push (_stream_readable.js:134:10) at TCP.onread (net.js:559:20)"
}

Executed 9 out of 13 tests successfully.
```

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Pull request merged**